File: Problems_on_Arrays/MaximumProductSubarray.java
package Problems_on_Arrays;

import java.util.Scanner;

public class MaximumProductSubarray {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the number of elements in the array:");
        int n = sc.nextInt();
        int[] arr = new int[n ];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        int max=Integer.MIN_VALUE;
        for (int i=0;i<n;i++){
            int tempmax=arr[i];
            for (int j=i+1;j<n;j++){
                tempmax*=arr[j];
                max=Math.max(max,tempmax);
            }
            
        }
        System.out.println("maximum product subarray is "+max);
        sc.close();
    }
    
}

File: Problems_on_Arrays/MaximumProductSubarray.py
n=int(input("enter the number of element :"))
arr=list(map(int,input("enter the elements :").split()))

max_prod=arr[0]

for i in range(n):
    prod=1
    for j in range(i,n):
        prod*=arr[j]
        if prod>max_prod:
            max_prod=prod

print("maximum product of subset is :",max_prod)